src/data_type.cpp

```diff
@@ -383,68 +385,39 @@ namespace cass {
-  DataTypeMap::const_iterator i = by_class_names_.find(name);
-  if (i == by_class_names_.end()) return DataType::NIL;
-  return i->second;
-}
-
-void NativeDataTypes::init_cql_names() {
-  if (!by_cql_names_.empty()) return;
-  by_cql_names_["ascii"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_ASCII));
-  by_cql_names_["bigint"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_BIGINT));
-  by_cql_names_["blob"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_BLOB));
-  by_cql_names_["boolean"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_BOOLEAN));
-  by_cql_names_["counter"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_COUNTER));
-  by_cql_names_["date"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_DATE));
-  by_cql_names_["decimal"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_DECIMAL));
-  by_cql_names_["double"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_DOUBLE));
-  by_cql_names_["duration"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_DURATION));
-  by_cql_names_["float"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_FLOAT));
-  by_cql_names_["inet"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_INET));
-  by_cql_names_["int"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_INT));
-  by_cql_names_["smallint"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_SMALL_INT));
-  by_cql_names_["time"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_TIME));
-  by_cql_names_["timestamp"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_TIMESTAMP));
-  by_cql_names_["timeuuid"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_TIMEUUID));
-  by_cql_names_["tinyint"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_TINY_INT));
-  by_cql_names_["text"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_TEXT));
-  by_cql_names_["uuid"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_UUID));
-  by_cql_names_["varchar"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_VARCHAR));
-  by_cql_names_["varint"] = DataType::ConstPtr(new DataType(CASS_VALUE_TYPE_VARINT));
-}
-
-const DataType::ConstPtr& NativeDataTypes::by_cql_name(const std::string& name) const {
-  DataTypeMap::const_iterator i = by_cql_names_.find(name);
-  if (i == by_cql_names_.end()) return DataType::NIL;
-  return i->second;
+DataType DataType::native_types_[CASS_VALUE_TYPE_LAST_ENTRY];
+
+struct DataTypeInitializer {
+  DataTypeInitializer() {
+    // Add a reference so that the memory is never deleted
+#define XX_VALUE_TYPE(name, type, cql) (new(&DataType::native_types_[name]) DataType(name))->inc_ref();
+    CASS_VALUE_TYPE_MAPPING(XX_VALUE_TYPE)
+    #undef XX_VALUE_TYPE
+  }
+};
+
+DataTypeInitializer __data_type_initializer__;
+
+DataType::ConstPtr DataType::get_native_by_class(const std::string& name) {
+  ValueTypeByClassMapping* mapping =
+    ValueTypeByClass::in_word_set(name.c_str(), name.length());
+  if (mapping == NULL) return DataType::NIL;
+  return DataType::ConstPtr(&native_types_[mapping->value_type]);
```

Comment:
The memory is statically allocated and initialized with "placement new". There's nothing to deallocated.